pyeudiw/trust/handler/x509.py

```diff
@@ -0,0 +1,77 @@
+import logging
+from typing import Union
+from pyeudiw.trust.handler.interface import TrustHandlerInterface
+from pyeudiw.trust.model.trust_source import TrustSourceData, TrustParameterData
+from pyeudiw.trust.handler.exceptions import InvalidTrustHandlerConfiguration
+from pyeudiw.x509.verify import (
+    verify_x509_attestation_chain, 
+    get_expiry_date_from_x5c, 
+    der_list_to_pem_list, 
+    pem_list_to_der_list, 
+    get_leaf_x509_dns_name,
+    get_root_x509_dns_name
+)
+
+logger = logging.getLogger(__name__)
+
+class X509Hanlder(TrustHandlerInterface):
+    """
+    X509Handler is a trust handler implementation that extracts trust material from x509 certificates.
+    """
+    def __init__(
+        self, 
+        client_id: str, 
+        relying_party_certificate_chains_by_ca: dict[str, Union[list[bytes], list[str]]],
+        private_keys: list[dict[str, str]],
+        **kwargs
+    ):  
+        if not relying_party_certificate_chains_by_ca:
+            raise InvalidTrustHandlerConfiguration("No x509 certificate chains provided")
+
+        self.relying_party_certificate_chains_by_ca = {}
+
+        for k, v in relying_party_certificate_chains_by_ca.items():
+            leaf_dns_name = get_leaf_x509_dns_name(v)
+            root_dns_name = get_root_x509_dns_name(v)
+            
+            if not root_dns_name in k:
```

Comment:
NO

CA and leaves must not necessarly belong to the same domain

ca.example.com can be the CA for leaf.example.eud and rp.acme.it